app/src/main/kotlin/com/wire/android/ui/common/topappbar/AppTopBarWithSearchBarLayout.kt

```diff
@@ -0,0 +1,277 @@
+                    modifier = Modifier
+                        .wrapContentHeight()
+                        .fillMaxWidth()
+                        .background(MaterialTheme.wireColorScheme.background)
+                ) {
+                    val interactionSource = remember {
+                        MutableInteractionSource()
+                    }
+
+                    val focusManager = LocalFocusManager.current
+
+                    SearchBarInput(
+                        placeholderText = searchBarHint,
+                        text = searchQuery,
+                        onTextTyped = onSearchQueryChanged,
+                        leadingIcon = {
+                            AnimatedContent(isTopBarVisible) { isVisible ->
+                                if (isVisible) {
+                                    IconButton(onClick = { }) {
+                                        Icon(
+                                            painter = painterResource(id = R.drawable.ic_search_icon),
+                                            contentDescription = stringResource(R.string.content_description_conversation_search_icon),
+                                            tint = MaterialTheme.wireColorScheme.onBackground
+                                        )
+                                    }
+                                } else {
+                                    IconButton(onClick = {
+                                        focusManager.clearFocus()
+
+                                        onCloseSearchClicked()
+                                    }) {
+                                        Icon(
+                                            painter = painterResource(id = R.drawable.ic_arrow_left),
+                                            contentDescription = stringResource(R.string.content_description_conversation_search_icon),
+                                            tint = MaterialTheme.wireColorScheme.onBackground
+                                        )
+                                    }
+                                }
+                            }
+                        },
+                        placeholderTextStyle = if (isTopBarVisible) LocalTextStyle.current.copy(textAlign = TextAlign.Center) else LocalTextStyle.current.copy(
+                            textAlign = TextAlign.Start
+                        ),
+                        textStyle = if (isTopBarVisible) LocalTextStyle.current.copy(textAlign = TextAlign.Center) else LocalTextStyle.current.copy(
+                            textAlign = TextAlign.Start
+                        ),
+                        interactionSource = interactionSource,
+                        modifier = Modifier.padding(8.dp)
+                    )
+
+                    if (interactionSource.collectIsPressedAsState().value) {
+                        onInputClicked()
+                    }
+                }
+            }
+        }
```

Comment:
The nesting level is getting too deep.

Does it make sense to extract some of the deeper elements to another @Compose?

Reply:
honestly I think this is pretty normal nesting with Composable, but will try to somehow exclude some stuff